providers/implementations/ciphers/cipher_aes_ocb.c

```diff
@@ -387,7 +387,10 @@ static int aes_ocb_set_ctx_params(void *vctx, const OSSL_PARAM params[])
         /* IV len must be 1 to 15 */
         if (sz < OCB_MIN_IV_LEN || sz > OCB_MAX_IV_LEN)
             return 0;
-        ctx->base.ivlen = sz;
+        if (ctx->base.ivlen != sz) {
+            ctx->base.ivlen = sz;
```

Comment:
This would only work if it was called before aes_ocb_init()? (Since update_iv() only works with IV_STATE_BUFFERED)

Reply:
Yes, setting **a different** IV length must work only if you call aes_ocb_init() setting a new IV. Other cases must fail because the IV is effectively unset after you change the IV length.